charts/castai-pod-pinner/templates/secret.yaml

```
@@ -10,7 +10,9 @@ metadata:
     {{- include "pod-pinner.annotations" . | nindent 4 }}
   {{- end }}
 data:
+  {{- if or .Values.castai.apiKey }}
```

This `or` is probably redundant here.

```suggestion
  {{- if .Values.castai.apiKey }}
```